db-auth-plan.md

```diff
@@ -224,45 +242,51 @@ CREATE TABLE oauth_authorization_request (
     code VARCHAR(64) PRIMARY KEY,              -- The authorization code
     entity_id INT NOT NULL,                    -- User who authorized
 
-    -- PKCE
-    code_challenge VARCHAR(128) NOT NULL,      -- SHA256 hash of verifier
-    code_challenge_method VARCHAR(10) NOT NULL, -- "S256" or "plain"
+    -- PKCE (only S256 supported)
+    code_challenge VARCHAR(128) NOT NULL,      -- BASE64URL(SHA256(code_verifier))
 
     -- Request details
     redirect_uri TEXT NOT NULL,
     app_name VARCHAR(255),
-    requested_scope TEXT NOT NULL,             -- JSON: {"database": "marcua/todos", "permission": "read-write"}
+    requested_query_permission_level SMALLINT NOT NULL, -- What the app requested (QueryMode value)
     state VARCHAR(255),                        -- Passed through to redirect
 
-    -- Selected by user
-    database_id INT,                           -- The database user selected
-    query_permission_level SMALLINT,           -- The permission level user approved (QueryMode values)
+    -- Selected by user during authorization
+    database_id INT NOT NULL,                  -- The database user selected or created
+    query_permission_level SMALLINT NOT NULL,  -- The permission level user approved (may be <= requested)
 
     created_at TIMESTAMP NOT NULL DEFAULT CURRENT_TIMESTAMP,
     expires_at TIMESTAMP NOT NULL,             -- Code expires after 10 minutes
-    used BOOLEAN NOT NULL DEFAULT FALSE,
+    used_at TIMESTAMP,                         -- NULL if unused, timestamp when exchanged for token
 
     FOREIGN KEY(entity_id) REFERENCES entity(id),
     FOREIGN KEY(database_id) REFERENCES database(id)
 );
 ```
 
+**Notes:**
+- Only S256 (SHA-256) is supported for `code_challenge` - the `plain` method is insecure and not implemented
+- `used_at` is NULL until the code is exchanged for a token, then set to the exchange timestamp
+- Codes with non-NULL `used_at` or past `expires_at` are invalid
+
 #### 2.2 New API Endpoints
 
-**GET `/oauth/authorize`** - Start authorization flow (redirects to UI)
+These endpoints live in `src/server/api_endpoints/` under the `/v1/` prefix.
+
+**GET `/v1/oauth/authorize`** - Start authorization flow (redirects to UI)
```

Comment:
There's something strange about an API endpoint that redirects to a UI endpoint. Is this standard? Should these get parameters simply go on the UI endpoint, which should forward you to login/registration if you're not logged in? Is there a security issue with that?